packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -1,83 +1,79 @@
-import React, { useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from 'react';
-import uPlot, { AlignedData, AlignedDataWithGapTest } from 'uplot';
+import React, { useCallback, useLayoutEffect, useMemo, useRef } from 'react';
+import uPlot, { AlignedData, AlignedDataWithGapTest, Options } from 'uplot';
 import { buildPlotContext, PlotContext } from './context';
 import { pluginLog } from './utils';
 import { usePlotConfig } from './hooks';
-import { PlotProps } from './types';
-import { usePrevious } from 'react-use';
+import { AlignedFrameWithGapTest, PlotProps } from './types';
 import { DataFrame, FieldType } from '@grafana/data';
 import isNumber from 'lodash/isNumber';
 import { UPlotConfigBuilder } from './config/UPlotConfigBuilder';
+import usePrevious from 'react-use/lib/usePrevious';
 
 // uPlot abstraction responsible for plot initialisation, setup and refresh
 // Receives a data frame that is x-axis aligned, as of https://github.com/leeoniya/uPlot/tree/master/docs#data-format
 // Exposes contexts for plugins registration and uPlot instance access
 export const UPlotChart: React.FC<PlotProps> = props => {
   const canvasRef = useRef<HTMLDivElement>(null);
-  const [plotInstance, setPlotInstance] = useState<uPlot>();
+  const plotInstance = useRef<uPlot>();
   const plotData = useRef<AlignedDataWithGapTest>();
-  const previousConfig = usePrevious(props.config);
-
-  // uPlot config API
-  const { currentConfig, registerPlugin } = usePlotConfig(props.width, props.height, props.timeZone, props.config);
-
-  const initializePlot = useCallback(() => {
-    if (!currentConfig || !plotData) {
-      return;
-    }
-    if (!canvasRef.current) {
-      throw new Error('Missing Canvas component as a child of the plot.');
-    }
-
-    pluginLog('UPlotChart: init uPlot', false, 'initialized with', plotData.current, currentConfig);
-    const instance = new uPlot(currentConfig, plotData.current, canvasRef.current);
-
-    setPlotInstance(instance);
-  }, [setPlotInstance, currentConfig]);
-
+  const prevProps = usePrevious(props);
+  const prevBuilder = usePrevious(props.config);
```

Comment:
this seems unnecessary, we can just use prevProps.config   (usePrevious is a bit expensive, it includes its own useEffect)

Reply:
great catch!